src/CoreWCF.Primitives/src/CoreWCF/Security/MessageSecurityOverHttp.cs

```diff
@@ -34,16 +34,6 @@ public MessageCredentialType ClientCredentialType
                     throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(value)));
                 }
 
-                if(value is MessageCredentialType.Windows)
-                {
```

Comment:
You can't remove this check yet. The Windows auth implementation depends on reflecting over an internal platform abstraction layer implementation in .NET Core which works on Windows and Linux. That internal implementation does not exist on .NET Framework so we throw to prevent the reflection based code from being hit when running on .NET Framework.